tests/integration/Core/Content/Category/SalesChannel/NavigationRouteTest.php

```diff
@@ -239,6 +242,175 @@ public function testInclude(): void
         static::assertArrayNotHasKey('id', $response[0]);
     }
 
+    /**
+     * @return array<string, array{linkType: string, categoryId: string, targetId: ?string, expectedUrlPart: string, createEntity: bool, preConfiguredSeoUrl: bool}>
+     */
+    public static function internalLinkSeoUrlDataProvider(): array
```

Comment:
ok hold on a second the data provider is definitely not easier to understand then multiple tests, can you go back to having one test method per link type and in every test method first test the result without a configured SEO URL , then add the SEO URL and check again that the response changed as expected?

With that extract common code you need in all test methods to provide methods instead of copying the code to every test method